docs/tests.md

```diff
@@ -0,0 +1,22 @@
+# Testing
+To test our charts we have a 2 part testing setup. The first part is linting, charts get linted using the [chart-testing tool](https://github.com/helm/chart-testing) which lints all YAML files as well as checks if they are compliant to Helm's standards.
+The second part are acceptance tests which are ran by by installing the chart on a Minikube Kubernetes cluster using set values, these are then tested by a pod deployment running tests.
+
+## Acceptance tests
+Tests are stored in `<chart name>/templates/tests`, these are Helm templates deployed to the cluster on test. These consist of a Pod object which will be ran on test, this pod contains the setup for the test. To mark the test as success or fail Helm will check the Pod's exit status. The test can also contain extra resources like configmaps (to store the tests in).
+Since running these tests require installing the charts in a cluster an appropriate values file is needed. This can be one or multiple, they are stored in `<chart name>/ci` and have to end on `-values.yaml`. These are also used during linting to check the validity of the Helm template output.
+
+## Writing tests
+For charts with required values a `<name>-values.yaml` file is needed in `<chart name>/ci`. This is also a requirement for chart linting. These values are used to install the chart inside the cluster.
+The tests themselves are Kubernetes resources and are placed in `<chart name>/templates/tests`, they follow the Helm templating system and have access to all values. The tests are ran inside a Pod resource, this Pod will be checked for exit status. This pod will run in the same environment as where the chart is deployed.
```

Comment:
```suggestion
The tests themselves are Kubernetes resources and are placed in `<chart name>/templates/tests`, they follow the Helm templating system and have access to all values. The tests are run inside a Pod resource and it will be checked for exit status. This pod will run in the same environment where the chart is deployed.
```